Implemented Translation system

justinkluever opened the pull request:
The ARCLib translation system should be implemented now,
please give feedback if something is wrong or if something can be better.

Changed:
* ~The user cant send empty messages anymore (can be used to spam) (26f544a and 8edc7e0)~
* The Beta-Warning can be disabled now by the server admin (enabled by default) (eab2dbe)

Fixed:
* ~NULL entity if you switch to the phone and die while the timer runs (6c11923)~

I also created an PR for the translations on your translations repo:
https://github.com/ARitz-Cracker/aritzcracker-addon-translations/pull/34

Fixes where moved to PR #4 so it can be merged without the translation part i need to work on (but cant currently because of some problems)

justinkluever committed:
Add option to disable Beta message

justinkluever committed:
Add Translation string for disable_beta_message

justinkluever committed:
Implement ARCLib translation system

This currently uses my own webserver to host the translation files,
This may change later.

justinkluever committed:
Now compatible with ARCLib translation stuff

Missing translations can be 'fixed' later.

justinkluever committed:
Fix "Tried to use a NULL ent" error

This error occours when the player dies
while or right after switching to the phone

justinkluever committed:
Block empty messages

justinkluever committed:
Also prevent empty messages on server

justinkluever committed:
Update Translation URL

justinkluever committed:
Update CONTRIBUTING

justinkluever committed:
I think you mean ARCPhone, not ARCSlots

ARitz-Cracker commented:
Hey, thanks for the PR.
I appreciate the bug (and documentation) fixes, but it seems like localization is only partially implemented.
I'd appreciate if you separate your work-in-progress localization implementation from your bugfixes, as I'm probably not going to accept such a PR until it replaces most of the static strings with localization references, so that way when users select a language, they won't be frustrated with the rest of the addon still being in english.

ARitz-Cracker commented on CONTRIBUTING.md:
```diff
@@ -1,12 +1,12 @@
+## ARCPhone design goals
+* ARCPhone should be as user friendly as possible
+* ARCPhone should be as configurable as possible _in-game_ without the need for configuration files:
 
```
👍

ARitz-Cracker commented on CONTRIBUTING.md:
```diff
@@ -1,12 +1,12 @@
 ## Things that I've always wanted to do but never got around to it.
 * **APP API NEEDS AN OVERHAULED AND MUST BE DOCUMENTED**
-* Make the addon compatible with the ARCLib translation system
+* ~~Make the addon compatible with the ARCLib translation system~~
```
I wouldn't call this complete just yet

justinkluever replied:
I will try to find and setup any string i find for the translation system asap

ARitz-Cracker commented on lua/arcphone/client/aa_phoneos_core.lua:
```diff
@@ -455,7 +455,9 @@ function ARCPhone.PhoneSys:Init_Final()
+	if not ARCPhone.Settings.disable_beta_message then
+		self:AddMsgBox("BETA VERSION","This is an unfinished BETA version of ARCPhone.\nMost of the core functionality is here, and it works, but there's still a bit more to go. Because of this, features might be added, changed, replaced, or even removed.\nThank you for supporting ARitz Cracker! :)\n(Press ENTER to close this window)","round-info-button")
+	end
 	--self:AddMsgBox("My excuse for a tutorial","Use the Arrow keys to move the cursor. Press BACKSPACE to go back, press CTRL to access the context menu (It's kinda like right-clicking), and press ENTER to select.","info")
```
Good addition, especially since you're not really supporting _me_ anymore

justinkluever replied:
Sorry if i did something wrong :eyes:
i didnt thought about that a "this is a beta" message reduce the support of the original creator in any way

but in my opinion something like this should be an option to disable for the server configurator

but if you dont agree on this i can revert this commit

ARitz-Cracker commented on lua/arcphone/client/syst_app_04_messaging.lua:
```diff
@@ -138,7 +138,7 @@ function APP:OpenConvo(num,previewText)
 		local msg = tile.App.Tiles[tile.App.SendIcon-1].TextInput
-		if tile.App.Phone:SendText(num,msg) then
+		if msg ~= "" and tile.App.Phone:SendText(num,msg) then
 			tile.App.Tiles[tile.App.SendIcon-1]:SetText("")
```
👍

ARitz-Cracker commented on lua/arcphone/server/aacore.lua:
```diff
@@ -443,6 +443,11 @@ function ARCPhone.AddToCall(caller,reciever)
+
+	if string.len( msg ) <= 0 then -- We may have bigger problems if it is less than zero
+		-- ARCPhone.SendTextMsg(fromnum,tonum,"AUTO REPLY:\nPlease dont send empty messages. :(")
+		return
```
👍

ARitz-Cracker commented on lua/arcphone/server/aacore.lua:
```diff
@@ -709,8 +714,12 @@ function ARCPhone.Load()
+		ARCLib.LoadDefaultLanguages("ARCPhone","https://raw.githubusercontent.com/JustPlayerDE/aritzcracker-addon-translations/master/default_arcphone_languages.json",function(langChoices)
+			ARCLib.AddonAddSettingMultichoice("ARCPhone","language",langChoices)
+			ARCLib.SetAddonLanguage("ARCPhone")
+		end)
```
See my other comment regarding translation implementation

ARitz-Cracker commented on lua/weapons/weapon_arc_phone.lua:
```diff
@@ -104,7 +104,7 @@ function SWEP:Deploy()
 	end
 	timer.Simple(1.5,function()
-		if self.Owner:GetActiveWeapon() == self then
+		if IsValid(self) and self.Owner:GetActiveWeapon() == self then
```
👍

justinkluever committed:
Update Contributing file

justinkluever committed:
Add cmds.lua Translations

justinkluever committed:
More correct AntennasRespawned instead of AntennasRespawn

justinkluever committed:
Add clcomm.lua Translations

justinkluever committed:
Add temporary TODO tag for translations

I'm using an extension to view these tags so i dont forget
to translate something later on

justinkluever commented:
I'm working now on the full-localization part

i added an `TODO: Translation` everywhere i found a string which i can add to the system
(i ignored server-console stuff for now)

should i have missed something im sorry ^^

justinkluever committed:
First client/amenu.lua translation patch

ARitz-Cracker commented:
You can ignore the server console stuff, I've always done that. Localization was always intended for the players or in-game admins. I always assumed that the person _installing_ the addons would know enough English to understand what gets put there.

justinkluever committed:
Fix wrong translation path (oops)

justinkluever commented:
Okay, thanks for the information :)

justinkluever committed:
Add missing translation for AdminMenu.ServerLogs
